pkg/trainer/replicas.go

```diff
@@ -492,7 +492,7 @@ func (s *TFReplicaSet) GetStatus() (spec.TfReplicaStatus, error) {
 }
 
 func (s *TFReplicaSet) jobName(index int32) string {
-	return fmt.Sprintf("%v-%v-%v", strings.ToLower(string(s.Spec.TfReplicaType)), s.Job.job.Spec.RuntimeId, index)
+	return fmt.Sprintf("%v-%v-%v-%v",fmt.Sprintf("%.16s", s.Job.job.Metadata.Name), strings.ToLower(string(s.Spec.TfReplicaType)), s.Job.job.Spec.RuntimeId, index)
```

Comment:
How did you arrive at 16 characters? I think the names have to be compliant with the DNS_LABEL spec since we use them as service names. 
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md
So I think it can be up to 63 characters.
Replica type currently has a max length of 6. 
Runtime id is 4 characters
Index is probably 4 characters

So that leaves about 46 characters. Maybe round down to 40 to leave 6 more characters for replicatype if needed.

Reply:
Because I don't want to see a too long job name. First I think of 8, but normally a job name could extend that. My first consideration is simplicity/clear.

Reply:
Just imagine seeing a pod name like xxxx(40 chars)-ps-xxxx-0. I don't think it good

Reply:
Thanks.   

Can you add a comment please describing the reasoning behind the format and including the link to identifiers.md.

If you don't want a long pod name why not just pick a short job name? 

I agree with you that long job names are awkward; but by setting the limit to a high number we leave it up to the user. Users who want short pod names can pick short job names.